src/elements/list/List.js

```diff
@@ -0,0 +1,121 @@
+import { Component } from 'angular2/core';
+import { CORE_DIRECTIVES, NgFor, NgIf, NgClass } from 'angular2/common';
+
+@Component({
+    selector: 'novo-list',
+    inputs: ['direction'],
+    host: {
+        '[class.vertical-list]': 'vert',
+        '[class.horizontal-list]': 'horiz'
+    },
+    template: `
+        <ng-content></ng-content>
+    `,
+    directives: [CORE_DIRECTIVES, NgFor]
+})
+export class NovoList {
+    constructor() {
+    }
+
+    ngOnInit() {
+        this.vert = (this.direction === 'vertical') ? true : null;
+        this.horiz = (this.direction === 'horizontal') ? true : null;
+    }
+}
+
+@Component({
+    selector: 'novo-list-item',
+    template: `
+        <ng-content select="item-avatar"></ng-content>
+        <div>
+            <ng-content select="item-title"></ng-content>
+            <ng-content select="item-content"></ng-content>
+        </div>
+        <ng-content select="item-end"></ng-content>
+    `,
+    directives: [CORE_DIRECTIVES, NgFor, NgIf]
+})
+export class NovoListItem {
+    constructor() {
+    }
+
+    ngOnInit() {
+    }
+}
+
+@Component({
+    selector: 'item-avatar',
+    inputs: ['icon'],
+    host: {},
+    template: `
+        <i *ngIf="iconClass" [ngClass]="classMap" theme="contained"></i>
+    `,
+    directives: [NgIf, NgClass]
+})
+export class ItemAvatar {
+    constructor() {
```

Comment:
Get rid of empty constructor, then you can move the logic to `ngOnChanges` and in `ngOnInit` call `this.ngOnChanges()` and you will be able to change the icon on the fly